pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -559,8 +559,9 @@ public ServerReloadControllerJobStatusResponse getReloadJobStatus(
               + controllerJobZKMetadata.get(CommonConstants.ControllerJob.SUBMISSION_TIME_MS);
       if (segmentNames != null) {
         List<String> targetSegments = serverToSegments.get(server);
-        reloadTaskStatusEndpoint = reloadTaskStatusEndpoint + "&segmentName=" + StringUtils.join(targetSegments,
-            SegmentNameUtils.SEGMENT_NAME_SEPARATOR);
+        List<String> encodedTargetSegments = targetSegments.stream().map(URIUtils::encode).collect(Collectors.toList());
```

Comment:
(minor) Consider looping over `targetSegments` and manually construct the string to avoid extra memory overhead